lib/ansible/modules/user.py

```diff
@@ -1250,18 +1233,26 @@ def create_homedir(self, path):
                     os.makedirs(path)
                 except OSError as e:
                     self.module.exit_json(failed=True, msg="%s" % to_native(e))
-            # get umask from /etc/login.defs and set correct home mode
-            if os.path.exists(self.LOGIN_DEFS):
-                with open(self.LOGIN_DEFS, 'r') as f:
-                    for line in f:
-                        m = re.match(r'^UMASK\s+(\d+)$', line)
-                        if m:
-                            umask = int(m.group(1), 8)
-                            mode = 0o777 & ~umask
-                            try:
-                                os.chmod(path, mode)
-                            except OSError as e:
-                                self.module.exit_json(failed=True, msg="%s" % to_native(e))
+            umask_string = None
+            # If an umask was set take it from there
+            if self.umask is not None:
+                umask_string = self.umask
+            else:
+                # try to get umask from /etc/login.defs
+                if os.path.exists(self.LOGIN_DEFS):
+                    with open(self.LOGIN_DEFS, 'r') as f:
+                        for line in f:
+                            m = re.match(r'^UMASK\s+(\d+)$', line)
+                            if m:
+                                umask_string = m.group(1)
```

Comment:
It should be ok to stop iterating through the file once we get a match.
```suggestion
                                umask_string = m.group(1)
                                break
```

Reply:
Sorry for being picky here, but this is not just an optimization, but also a change in behaviour. The old code takes the last value of `UMASK` in `/etc/login.defs` and the new one the first. I made a quick check on a Centos 8 system and at least `useradd` takes the *last* value of UMASK if there are multiple entries in `/etc/login.defs`. Hence I would propose to leave the code as is but add a comment that on purpose we don't do a break after a match as we want to get the last value just like `useradd`.

Reply:
That's a very good point. I haven't ever encountered an `/etc/login.defs` file with more than one `UMASK` definition so it's not something I thought of. We can just leave it as is.